src/progress.rs

```diff
@@ -50,6 +52,117 @@ impl<'a> Progress<'a> {
         )
         .await
     }
+
+    /// Wait until MeiliSearch processes an update, and get its status.
+    /// 
+    /// interval_ms = The frequency at which the server should be polled. Default = 50ms
+    /// timeout_ms = The maximum time to wait for processing to complete. Default = 5000ms
+    /// 
+    /// If the time waited excedes timeout_ms then None will be returned.
+    /// 
+    /// # Example
+    ///
+    /// ```
+    /// # use meilisearch_sdk::{client::*, document, indexes::*, progress::*};
+    /// # use serde::{Serialize, Deserialize};
+    /// #
+    /// # #[derive(Debug, Serialize, Deserialize, PartialEq)]
+    /// # struct Document {
+    /// #    id: usize,
+    /// #    value: String,
+    /// #    kind: String,
+    /// # }
+    /// # 
+    /// # impl document::Document for Document {
+    /// #    type UIDType = usize;
+    /// #
+    /// #    fn get_uid(&self) -> &Self::UIDType {
+    /// #        &self.id
+    /// #    }
+    /// # }
+    /// #
+    /// # futures::executor::block_on(async move {
+    /// let client = Client::new("http://localhost:7700", "masterKey");
+    /// let movies = client.create_index("movies_wait_for_pending", None).await.unwrap();
+    /// 
+    /// let progress = movies.add_documents(&[
+    ///     Document { id: 0, kind: "title".into(), value: "The Social Network".to_string() },
+    ///     Document { id: 1, kind: "title".into(), value: "Harry Potter and the Sorcerer's Stone".to_string() },
+    /// ], None).await.unwrap();
+    /// 
+    /// let status = progress.wait_for_pending_update(None, None).await.unwrap();
+    /// 
+    /// # client.delete_index("movies_wait_for_pending").await.unwrap();
+    /// assert!(matches!(status.unwrap(), UpdateStatus::Processed { .. }));
+    /// # });
+    /// ```
+    pub async fn wait_for_pending_update(
+        &self,
+        interval_ms: Option<Duration>,
+        timeout_ms: Option<Duration>,
+    ) -> Option<Result<UpdateStatus, Error>> {
+        let interval: Duration;
+        let timeout: Duration;
+
+        match interval_ms {
+            Some(v) => interval = v,
+            None => interval = Duration::from_millis(50),
+        }
+
+        match timeout_ms {
+            Some(v) => timeout = v,
+            None => timeout = Duration::from_millis(5000),
+        }
```

Comment:
```suggestion
        let interval = interval.unwrap_or(Duration::from_millis(50));
        let timeout = timeout.unwrap_or(Duration::from_millis(5000));
```

Reply:
@Mubelotix this update triggers a clippy warning ```use of `unwrap_or` followed by a function call```. I can resolve the warning by instead using

```rust
let interval = interval.unwrap_or_else(|| Duration::from_millis(50));
let timeout = timeout.unwrap_or_else(|| Duration::from_millis(5000));
```

Does this look good to you? All other suggestions are added and ready to go.

Reply:
Yes that's great